### ISSUE TITLE
"-inf" in floats causing issues

### ISSUE THREAD

**Issue by sg2nq** [google/draco#633]
Hi! I'm trying to compress a point cloud using draco_encoder CLI tool, and the PLY file I'm converting has a few `-inf`s. 
When I try to decode without using cmd line params for quantization bits, all values get set to `-inf`.
I'm open to going uncompressed for the float that is `-inf`, but if I try to make quantization to be set to 0 I get:
```
Failed to encode the point cloud.
Failed to encode point attributes.
```

**Comment by sg2nq** [comment]
I did a little more testing and learned that if you convert to doubles, it works fine. 
Alternately, if you convert `-inf`s to`nan`s, it also works fine - almost. It doesn't set everything to `-inf` but it doesn't let you set the quantization to `0`, same error as above

**Comment by igorvytyaz** [comment]
The `+/-inf` values are only supported for non-quantized attributes, so for your described example we expect encoding to work only with `-qp 0`. Are you able to encode this [simple example](https://github.com/google/draco/files/5351771/rename_to_ply.txt)? Please share your actual example if possible.

**Comment by sg2nq** [comment]
My particular situation is a little more complex. But in summary, I modified the CLI tool to add extra attributes (by using generic attributes, adding those in the ply_decoder.cc, almost identical to this block [here](https://github.com/google/draco/blob/bde99c26634f6f9d3944ea6b68782b77c791e0d6/src/draco/io/ply_decoder.cc#L280-L292). The kicker is that I want to use -qp at a non zero value (13 for instance), but I would be willing to use `-qg 0`. However, using `-qg 0` gives the error I added above, when using a -qp value eg. `-qp 13 -qg 0`. 
Using both `-qp 0 -qg 0` works perfectly, but that's not an acceptable solution for me. 
So to summarize, my two follow up questions are:
1. Why does it not work with quantized attributes? Can support for this situation be added. -inf is an easy way to invalidate a point, similar to `nan` (which leads into my second question)
2. Why can't `nan` work with `-qg 0`? You can try recreating it, but having `-qp 13 -qg 0` gives me a Failed to encode error or a seg fault (I can't remember which).

**Comment by ondys** [comment]
In general, both NaN or Inf are not supported for point cloud encoding except if the affected attribute is not quantized and the point cloud encoding method is set to POINT_CLOUD_SEQUENTIAL_ENCODING (so -qp 13 -qg 0 should work as long as you set the encoding method to POINT_CLOUD_SEQUENTIAL_ENCODING .. assuming positions don't have any inf/nan values)

If you want to use nan or inf with quantized attributes I would suggest doing quantization / dequantization yourself outside of Draco compression (Draco can compress integer "pre-quantized" attributes). You can then reserve certain integer values for nan or inf values. This is a special case that we do not plan to handle natively within the codec itself.

**Comment by sg2nq** [comment]
Ok, thank you, you can consider this "bug" fixed then. I do recommend, however, that you try out setting "-qp 13 - qg 0" with a generic attribute float32. I believe it'll cause issues, even if there are no special values (nan, inf).

**Comment by igorvytyaz** [comment]
Thank you for this helpful discussion @sg2nq. The next Draco release will have two related enhancements:
- Error will be reported when trying to quantize `inf` and `nan` values.
- Mix of quantized and unquantized float attributes will be supported. Meanwhile, you could make a fix in your local code by replacing `GetAttributeInt(0, ...)` with `GetAttributeInt(i, ...)` [here](https://github.com/google/draco/blob/bde99c26634f6f9d3944ea6b68782b77c791e0d6/src/draco/compression/expert_encode.cc#L70).

**Comment by sg2nq** [comment]
Awesome, thank you!

**Comment by FrankGalligan** [comment]
Closing as 1.4.0 has been released.
